### PR TITLE
moves guice into it's own module and updates/migrates the README

### PR DESCRIPTION
**Pull request by schmitch**
guice will now be inside it's own module
README will now be converted to markdown to have the same
doc language in all projects

### CODE REVIEW

**Comment by mkurz** [comment]
@schmitch Please rebase, thanks

**Comment by schmitch** [comment]
rebased

File: README.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+  timeout = null // (defaults to 60s in milliseconds)
+  connectiontimeout = null // (defaults to 60s in milliseconds)
+  mock = no // (defaults to no, will only log all the email properties instead of sending an email)
+}
+```
+
+If you are on Play 2.6.x you can skip to [Usage](#Usage).
+For Play 2.5.x you might also need a ConfigModule:
+
+### Scala 2.5.x
```
Comment (marcospereira) [inline README.md]: "Scala 2.5.x" -> "Play 2.5.x"

Comment (marcospereira) [inline README.md]: Oh, got it. This is the Scala example.

Anyway, I think this title could be clearer: "Play 2.5.x and Scala".

File: README.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+play.modules {
+    # Disable the default provider
+    disabled += "play.api.libs.mailer.SMTPConfigurationModule"
+    # Enable the custom provider (see above)
+    enabled += "controllers.CustomMailerConfigurationModule"
+}
+```
+
+> The get() method of your CustomSMTPConfigurationProvider will be called multiple times.
+> s a consequence, we recommend that code inside the get() method should be fast.
```
Comment (marcospereira) [inline README.md]: Small typo here: `s a consequence...` -> `as a consequence...`.

File: build.sbt [hunk 3]
```diff
@@ -16,21 +17,39 @@ lazy val commonSettings = SbtScalariform.scalariformSettings ++ Seq(
...
-  "org.apache.commons" % "commons-email" % "1.4",
-  "com.typesafe.play" %% "play" % PlayVersion % Test,
-  "com.typesafe.play" %% "play-specs2" % PlayVersion % Test
-)
+lazy val `play-mailer` = (project in file("play-mailer"))
+  .enablePlugins(PlayLibrary)
+  .settings(commonSettings)
+  .settings(
+    libraryDependencies ++= Seq(
+      "javax.inject" % "javax.inject" % "1", // 4.0 to maybe make it work with 2.5 and 2.6
```
Comment (mkurz) [inline build.sbt]: The `// 4.0...` comment can be removed

**Pull request merged**